### PR TITLE
Correctly create power level event during initial room creation

### PR DESCRIPTION
**Pull request by H-Shay**
#14228 added the initial power level event to the batched events but the event was not created for batching. This fixes that.

### CODE REVIEW

File: changelog.d/14361.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix a bug introduced in #14228 where the power level event was incorrectly created during initial room creation.
```
Comment (reivilibre) [inline changelog.d/14361.bugfix:1]: was the bug ever released?
If so, cite the version number instead.
If not, probably just coalesce the changelogs.

File: synapse/handlers/room.py [hunk 2]
```diff
@@ -1139,13 +1139,21 @@ async def create_event(
...
+        current_state_group = event_to_state[member_event_id]
+
         events_to_send = []
         # We treat the power levels override specially as this needs to be one
         # of the first events that get sent into a room.
         pl_content = initial_state.pop((EventTypes.PowerLevels, ""), None)
         if pl_content is not None:
             power_event, power_context = await create_event(
-                EventTypes.PowerLevels, pl_content, False
+                EventTypes.PowerLevels, pl_content, True
```
Comment (reivilibre) [inline synapse/handlers/room.py:1156]: this bool seems to be `for_batch`, which doesn't have a docstring.
Would you mind adding a docstring on it? I would find it especially useful to know why it's needed and what the semantics are — from the name I'm guessing the usage is just `True when you are creating an event to send in a batch`, but the other parts are more interesting

**Review by reivilibre — CHANGES_REQUESTED** [review]

File: synapse/handlers/room.py (hunk 2 above)
Comment (H-Shay) [reply synapse/handlers/room.py:1163]: I've added a docstring, let me know if it makes more sense.

File: changelog.d/14361.bugfix (hunk 1 above)
Comment (H-Shay) [reply changelog.d/14361.bugfix]: Yes, it was released in most recent rc. I've updated the changelog.

Comment (reivilibre) [reply changelog.d/14361.bugfix]: Ah, is it serious enough to go into the release branch?
Actually it occurs to me that I don't understand what 'the power level event was incorrectly created' means — at first thought this sounds like the power levels event wasn't meant to be created, but surely that can't be right.

Iit would be best to note the user-visible bug if there is one. I honestly have lost track of this over the weekend so I'm afraid I can't remember enough to advise.

File: synapse/handlers/room.py (hunk 2 above)
Comment (reivilibre) [reply synapse/handlers/room.py:1163]: makes sense, but maybe would benefit from an elaboration about what the difference is

File: changelog.d/14361.bugfix (hunk 1 above)
Comment (H-Shay) [reply changelog.d/14361.bugfix]: There isn't an end-user facing bug afaict. From my understanding, the worst thing that could happen with this bug is the code could break unexpectedly if other changes were made to the surrounding code, so mostly fixing this is to a: prevent confusion in the future and b: set up the code for further changes (such as adding the membership event to the batch or batching up state groups).

**Comment by H-Shay** [comment]
Merged #14361 into develop.

**Pull request merged**